description/src/lib.rs

```diff
@@ -46,61 +87,79 @@ mod tests {
             }"}
         );
     }
-    // todo!("This test with the generics does not fly yet.")
-    // #[test]
-    // fn enums() {
-    //     #[allow(unused)]
-    //     #[derive(TypeInfo)]
-    //     enum Shape<T> {
-    //         Inivisible,
-    //         Circle(u64),
-    //         Rect(Compact<u64>, Compact<u64>),
-    //         Polygon {
-    //             corners: u8,
-    //             radius: u64,
-    //         },
-    //         MultiShape {
-    //             shapes: Vec<Shape<u32>>,
-    //             t: T,
-    //             operation: Operation,
-    //         },
-    //     }
-
-    //     #[allow(unused)]
-    //     #[derive(TypeInfo)]
-    //     enum Operation {
-    //         Add,
-    //         Intersect,
-    //         Difference,
-    //     }
-
-    //     let (type_id, type_registry) = make_type::<Shape<bool>>();
-
-    //     assert_eq!(
-    //         type_description(type_id, &type_registry).unwrap(),
-    //         indoc! {
-    //         "enum Shape {
-    //             Inivisible,
-    //             Circle(u64),
-    //             Rect(
-    //                 Compact<u64>,
-    //                 Compact<u64>
-    //             ),
-    //             Polygon  {
-    //                 corners: u8,
-    //                 radius: u64
-    //             },
-    //             MultiShape  {
-    //                 shapes: Vec<Shape>,
-    //                 operation: enum Operation {
-    //                     Add,
-    //                     Intersect,
-    //                     Difference
-    //                 }
-    //             }
-    //         }"}
-    //     );
-    // }
+
+    #[test]
+    fn enums() {
+        use parity_scale_codec::Compact;
+
+        #[allow(unused)]
+        #[derive(TypeInfo)]
+        enum Shape<T> {
+            Inivisible,
+            Circle(u64),
+            Rect(Compact<u64>, Compact<u64>),
+            Polygon {
+                corners: u8,
+                radius: u64,
+            },
+            MultiShape {
+                shapes: Vec<Shape<u64>>,
+                t: T,
+                operation: Operation,
+            },
+        }
+
+        #[allow(unused)]
+        #[derive(TypeInfo)]
+        enum Operation {
+            Add,
+            Intersect,
+            Difference,
+        }
+
+        let (type_id, type_registry) = make_type::<Shape<u8>>();
+        assert_eq!(
+            type_description(type_id, &type_registry, true).unwrap(),
+            indoc! {
+            "enum Shape {
+                Inivisible,
+                Circle(u64),
+                Rect(Compact<u64>, Compact<u64>),
+                Polygon  {
```

Comment:
Nit; why two spaces between variant name and `{` here and below?

Reply:
That was a bug - fixed it.